types/sharedb/index.d.ts

```diff
@@ -93,6 +103,20 @@ declare class sharedb extends EventEmitter {
     on(event: 'timing', callback: (type: string, time: number, request: any) => void): this;
     on(event: 'submitRequestEnd', callback: (error: Error, request: SubmitRequest) => void): this;
     on(event: 'error', callback: (err: Error) => void): this;
+    on(event: 'send', callback: (agent: Agent, context: SubmitContext
```

Comment:
I don't believe the type of the second event argument here is accurate.

In the implementation, the second arg is the raw JSON message being sent to the client, not a middleware Context:
https://github.com/share/sharedb/blob/a837b0090ef4731ca4605d067a40a61e6ef0410c/lib/agent.js#L247

This is how most of the JSON response messages are structured:
https://github.com/share/sharedb/blob/a837b0090ef4731ca4605d067a40a61e6ef0410c/lib/agent.js#L303-L312

Reply:
Makes sense. Will test some more and come up with a more accurate definition.

The raw JSON message is why I have the `{[key: string]: any}` type, but I need to see if there is a more concrete type we can use.

Reply:
Testing with adding an error in sharedb that throws 50% of the time, we see the following signatures for the structure of the second argument in the "send" event:

```
a, c, d, error, v {"a":"s","c":"user_settings","d":"some-uuid","v":20932,"error":{"message":"some-error"}}
derby {"derby":"reload"}
a, c, d {"a":"s","c":"accounts","d":"some-uuid"}
a, b, c {"a":"bs","c":"users_base","b":{"some-uuid":165672}}
a, id {"a":"qs","id":3}
a, extra, id {"a":"q","id":14,"extra":0}
```